compiler/rustc_typeck/src/check/mod.rs

```diff
@@ -988,3 +991,37 @@ fn has_expected_num_generic_args<'tcx>(
         generics.count() == expected + if generics.has_self { 1 } else { 0 }
     })
 }
+
+/// Suggests calling the constructor of a tuple struct or enum variant
+///
+/// * `snippet` - The snippet of code that references the constructor
+/// * `span` - The span of the snippet
+/// * `params` - The number of parameters the constructor accepts
+/// * `err` - A mutable diagnostic builder to add the suggestion to
+fn suggest_call_constructor<G: EmissionGuarantee>(
+    snippet: impl std::fmt::Display,
```

Comment:
To do what I proposed, you'd change this to pass the `expr.span` instead of the `expr_snippet`.